File: platforms/web/examples/texture/texture/write_region.js
import { Renderer, TextureFormat } from "fragmentcolor";

const renderer = new Renderer();
const texture = await renderer.createStorageTexture([64, 32], TextureFormat.Rgba, null);
const bytes = new Uint8Array(64 * 32 * 4);

// Simple sub-rectangle update.
texture.writeRegion(bytes, [0, 0, 64, 32]);

// Explicit data layout (advanced).
texture.writeRegion(bytes, { x: 0, y: 0, width: 64, height: 32, bytesPerRow: 256, rowsPerImage: 32 });

File: platforms/python/examples/texture/texture/write_region.py
from fragmentcolor import Renderer, TextureFormat

renderer = Renderer()
texture = renderer.create_storage_texture([64, 32], TextureFormat.Rgba)
bytes_data = bytes(64 * 32 * 4)

# Simple sub-rectangle update.
texture.write_region(bytes_data, [0, 0, 64, 32])

# Explicit data layout (advanced).
texture.write_region(bytes_data, {"x": 0, "y": 0, "width": 64, "height": 32, "bytes_per_row": 256, "rows_per_image": 32})